### DISCUSSION TITLE
osg-satellite：Modify the connection method

### DISCUSSION THREAD

**Discussion by xpyeah** [omnetpp/omnetpp#1022] [Q&A] [answered]
I'm using the OSG-satellite module for satellite simulation. The satellite connection in the module makes use of the LineOfSight module (the module that comes with osgearth). I want to modify the connection method (using a custom connection algorithm), but I can't find the .cc file for that module.In other words, how should I introduce my connection algorithm

**Comment by rhornig** [comment]
This is not a bug. Please use the Discussion section for questions. Moving the issue.

**Comment by rhornig** [comment, marked as answer]
The osg-satellite example is a demo for showing visualization. It does not have ANY meaningful model for satellite orbit, line of sight calculation, or anything that is needed for a proper communication model. I do not recommend starting with it.

Once you have your model working, it's OK to use this to enhance the visualization of the model.

regarding your actual question: You do not find the line of sight calculation, because it is NOT in the model. The example uses some functions from the osgEarth library, that's why you cannot change it.
